src/dialogs/notifications/notification-drawer.js

```diff
@@ -122,11 +123,31 @@ export class HuiNotificationDrawer extends EventsMixin(
         this.hass.connection,
         (notifications) => {
           this._notificationsBackend = notifications;
+          this.addEventListener(
+            "notifications-changed",
+            this._onNotificationsChanged
+          );
```

Comment:
We can call this in `_computeNotifications` instead of with an event? And otherwise we should add an `observer` on `notifications` instead of the `notify`

Reply:
The issue with all approaches that don't include an event observer is that the notifications are loaded async. That means when the user tries to _open_ the drawer, for a split second `notifications.length === 0`, and the drawer immediately closes again (actually stays closed).
This is an issue both if we use an `observer` pattern (like `_openChanged` does), and if we move this into `_computeNotifications`. 

While encapsulating everything in `_computeNotifications` is a lot more compact, it runs into the issue mentioned above:
```JS
  _computeNotifications(open, hass, notificationsBackend) {
    if (!open) {
      return [];
    }

    const configuratorEntities = Object.keys(hass.states)
      .filter((entityId) => computeDomain(entityId) === "configurator")
      .map((entityId) => hass.states[entityId]);

    const notifications = notificationsBackend.concat(configuratorEntities);

    // this evaluates to true when the user tries to open the drawer before we have
    // received any notifications from the backend-subscription, thus preventing the drawer from opening
    if (notifications.length === 0) {
        this.open = false;
    }

    return notifications;
  }
```

The only way I have found to avoid this is by attaching the observer once we have loaded all notifications. Another option would be to rewrite the whole component to set up the server subscription in `PolymerElement.ready`, but that would be a larger refactor that I did not feel comfortable proposing for this small feature.

Reply:
We should rewrite it to a `LitElement`, that will solve your problem and should be done anyway

Reply:
If you want that challenge that is :-)

Reply:
I have not worked with Polymer or LitElement before, so I'll have to see how deep I want to go into that rabbit hole for adding this mini-feature. 😄 
Until then, is this PR not acceptable?

Reply:
I would not like to merge it as is, if you want to keep the Polymer you can use:
```js
  static get properties() {
    return {
	  ...,
      notifications: {
        type: Array,
        computed: "_computeNotifications(open, hass, _notificationsBackend)",
        observer: "_notificationsChanged",
      },
      ...,
    };
  }

  _notificationsChanged(newNotifications, oldNotifications) {
    if (this.open && oldNotifications.length && !newNotifications.length) {
      this.open = false;
    }
  }
```

Reply:
I wasn't aware of the `newNotifications, oldNotifications` arguments to the observer function. This is a much easier approach. Thanks for the suggestion!

I adjusted the PR and propose merging like this. If I feel up for it, I'll create a separate PR for a LitElement rewrite, but don't want to commit to it at this time.